osu.Game/Screens/OnlinePlay/OnlinePlayScreen.cs

```diff
@@ -132,7 +132,9 @@ public override void OnResuming(ScreenTransitionEvent e)
             this.ScaleTo(1, 250, Easing.OutSine);
 
             Debug.Assert(screenStack.CurrentScreen != null);
-            screenStack.CurrentScreen.OnResuming(e);
+
+            if (screenStack.CurrentScreen.IsCurrentScreen())
+                screenStack.CurrentScreen.OnResuming(e);
```

Comment:
<sup>okay i won't ask for tests - duly noted - but</sup> how can `screenStack.CurrentScreen` _not be current_? That sounds... contradictory?

And a follow up question to that is - does this guard make it possible for a screen on the child stack to never have its `On{Resuming,Suspending}()` called? Sorry for asking this if it is incredibly obvious but to me it very much isn't and there's not much of an explanation in the description.

Reply:
Definitely should have explained this one better.

What we're dealing with here is a nested screen stack that is non-`Present` at the time of push (the screen owning it is suspended).

In this scenario, if a screen is pushed to the nested stack it immediately becomes `CurrentScreen` but doesn't receive `OnEntered` until it is present again.

Meanwhile, when we resume the nested stack owner it immediately receives `OnResuming` and forwards that to the new nested screen before it is entered, breaking sequence expectations completely.

Reply:
The explanation is helpful, thanks - I do still think this fix reads too WTF in its current form to let it go without at least some inline commentary.

The sole reason that the fix as written here fixes anything is an implementation detail of `IsCurrentScreen()` - namely the [`getStack()`](https://github.com/ppy/osu-framework/blob/9206408d84d1db4b75cf5b0a7c71783db395eb3c/osu.Framework/Screens/IScreen.cs#L114-L125) method therein. Because the screen supplied to it hasn't fully loaded yet, it hasn't received its `Parent`, causing the `getStack()` method to fail to find its stack (even though its stack *already says that the screen is current via* `CurrentScreen`).

I'm not saying that we should fix that, because I'm also scared of touching that code, but we can probably make this a lot more explicit by retargeting the band-aid. Something like this, perhaps?

```suggestion
            // if a subscreen was pushed to the nested stack while this screen was not present, this path will proxy `OnResuming()`
            // to the subscreen before `OnEntering()` can even be called for the subscreen, breaking ordering expectations.
            // to work around this, do not proxy resume to screens that haven't loaded yet.
            if ((screenStack.CurrentScreen as Drawable)?.IsLoaded == true)
                screenStack.CurrentScreen.OnResuming(e);

```

Reply:
I get what you're going for, but I'm not sure that reads better. The comment is definitely welcome though.

Reply:
I mean if we're going to keep your fix, then I think the comment must grow even longer to contain some of the explanation above about *why* `IsCurrentScreen()` returns false for a `CurrentScreen`, which should be basically impossible.

To me the current code just reads terribly wrong. On the face of it, it should _never_ be permitted to have `screenStack.CurrentScreen.IsCurrentScreen()` evaluate to false. And it only works due to an implementation detail.

Reply:
Fair. I think I'm biased from hitting this same issue multiple times in the past and having a feeling that the `CurrentScreen` vs `IsCurrentScreen()` discrepancy is something not easily fixable.

Reply:
I'm not sure why this was applied only to `OnResuming()` and not the exact same guard in `OnSuspending()`. To speed proceedings up I applied it in `OnSuspending()` too in https://github.com/ppy/osu/pull/23661/commits/62d177440730e507ff3b080dd8c1c1eca66ea6bc, but please do double-check that you're OK with that.